### DISCUSSION TITLE
How to handle "/" route with a serverless function ?

### DISCUSSION THREAD

**Discussion by tdurand** [vercel/vercel#4455] [Help] [answered]
Hi, 

I'm trying to move lots of my past Express.js projects deployed with nowV1 to vercel now v2 and I'm struggling with a very simple thing right now

I've created a vercel project with `vercel dev`

I created a api directory with an index.js function

If I go to localhost:3000/api it works.. but I would like to handle the localhost:3000/ root, and currently it just displays the directory listing...

I found that the documentation is very good on how to deploy stuff with next.js, gabsy.. but very cryptic on how to do this simple thing.

Some help ?

Thibault

**Comment by idjevm** [comment]
Hi there, 

Have you tried manually setting a now.json configuration file? Well, I believe it has changed to vercel.json configuration.

A simple example from me here: 

This is my vercel.json file on the root of the project
```{
  "version": 2,
  "builds": [
    { "src": "/app.js", "use": "@now/node" }
  ],
  "routes": [
    {
      "src": "/",
      "dest": "/app.js",
      "methods": ["GET"]
    },
    { "src": "/api/books", "dest": "/app.js", "methods": ["GET","POST", "DELETE", "PUT"]},
    { "src": "/api/books/(.*)", "dest": "/app.js", "methods": ["GET","POST", "DELETE", "PUT"]}
  ],
  "env": {
    "MONGO_DB_URI": "@mongo_db_uri" 
  }
}
```

This worked for me, try to give it a shot, you can read more about it here -> https://vercel.com/docs/configuration#introduction/configuration-reference

Let me know if that helped!

**Reply by borisdedejski** [reply]
even without "builds" for me works fine.

**Reply by ctrlplusb** [reply]
Does this configuration work via `vercel dev` command for you all?

**Reply by tdurand** [reply]
with this config it does work for me

```json
{
  "version": 2,
  "routes": [
    {
      "src": "/",
      "dest": "/api/index.js",
      "methods": ["GET"]
    }
  ]
}
```

**Comment by tdurand** [comment]
Hi @idjevm 

Many thanks this did the trick:

```json
{
  "version": 2,
 "builds": [
    { "src": "/index.js", "use": "@now/node" }
  ],
  "routes": [
    {
      "src": "/",
      "dest": "/index.js",
      "methods": ["GET"]
    }
  ]
}
```

Should have this somewhere in the doc as an alternative to the api/ directory in my opinion

Best, 

Thibault

**Comment by tdurand** [comment, marked as answer]
Even simpler:

```json
{
  "version": 2,
  "routes": [
    {
      "src": "/",
      "dest": "/api/index.js",
      "methods": ["GET"]
    }
  ]
}
```
